langs/odin/Dockerfile

```diff
@@ -2,52 +2,62 @@ FROM alpine:3.20 AS builder
 
 RUN apk add --no-cache clang curl llvm-dev make
 
-RUN curl -L https://github.com/odin-lang/Odin/archive/refs/tags/dev-2024-11.tar.gz | tar xz
+ENV VER=2024-11
 
-RUN mv Odin-* /usr/local/tmp
+RUN curl -#L https://github.com/odin-lang/Odin/archive/refs/tags/dev-$VER.tar.gz \
+  | tar xz --directory /usr/local/lib --strip-components 1
 
-RUN ln -s /usr/local/tmp/base /usr/local/bin/base \
- && ln -s /usr/local/tmp/core /usr/local/bin/core
+RUN ln -s /usr/local/lib/base \
+          /usr/local/bin/base \
+ && ln -s /usr/local/lib/core \
+          /usr/local/bin/core \
+ && mkdir /usr/local/bin/vendor
 
-RUN mkdir /usr/local/bin/vendor
+WORKDIR /usr/local/lib
 
-WORKDIR /usr/local/tmp
+RUN make -j`nproc` release \
+ && strip /usr/local/lib/odin
 
-RUN ODIN_ROOT=/usr/local/bin make release -j `nproc`
+COPY odinwrapper.c /
+
+RUN gcc -Wall -Werror -Wextra -o /usr/bin/odinwrapper -s -static /odinwrapper.c
 
 FROM codegolf/lang-base
 
-COPY --from=0 /bin/cat /bin/rm /bin/sh                                /bin/
-COPY --from=0 /lib/ld-musl-x86_64.so.1 /lib/libz.so.1                 /lib/
-COPY --from=0 /usr/bin/clang /usr/bin/ld                              /usr/bin/
-COPY --from=0 /usr/lib/crt1.o                                         \
-              /usr/lib/crti.o                                         \
-              /usr/lib/crtn.o                                         \
-              /usr/lib/libbfd-2.42.so                                 \
-              /usr/lib/libc.so                                        \
-              /usr/lib/libclang-cpp.so.17                             \
-              /usr/lib/libctf.so.0                                    \
-              /usr/lib/libffi.so.8                                    \
-              /usr/lib/libgcc_s.so                                    \
-              /usr/lib/libgcc_s.so.1                                  \
-              /usr/lib/libjansson.so.4                                \
-              /usr/lib/libLLVM-17.so                                  \
-              /usr/lib/liblzma.so.5                                   \
-              /usr/lib/libm.a                                         \
-              /usr/lib/libsframe.so.1                                 \
-              /usr/lib/libssp_nonshared.a                             \
-              /usr/lib/libstdc++.so.6                                 \
-              /usr/lib/libxml2.so.2                                   \
-              /usr/lib/libzstd.so.1                                   /usr/lib/
+COPY --from=0 /bin/sh                     /bin/
```

Comment:
This particular line annoys me. I'm not sure if this keeps that injection crap gate open, because I don't really know how it actually truly works. My interpretation is that if it's there, it's available and yet still problematic. Unfortunately, the language refuses to work for shit without its presence, and this is just one of multiple languages that depend on it.